terraform/modules/workflows_cf/main.tf

```diff
@@ -52,43 +27,38 @@ resource "google_service_account" "workflows_service_account" {
   depends_on = [google_project_service.iam]
 }
 
-# Grant the logWriter role to the service account
-resource "google_project_iam_binding" "project_binding_eventarc" {
-  provider = google-beta
-  project  = var.project
-  role     = "roles/logging.logWriter"
-
-  members = ["serviceAccount:${google_service_account.workflows_service_account.email}"]
-
-  depends_on = [google_service_account.workflows_service_account]
+resource "google_project_iam_member" "roles" {
+  for_each = toset(local.workflows_sa_roles)
+  project = var.project
+  role    = each.key
+  member  = "serviceAccount:${google_service_account.workflows_service_account.email}"
 }
 
-# Grant the workflows.invoker role to the service account
-resource "google_project_iam_binding" "project_binding_workflows" {
-  provider = google-beta
+resource "google_project_iam_member" "token-creator-iam" {
   project  = var.project
-  role     = "roles/workflows.invoker"
-
-  members = ["serviceAccount:${google_service_account.workflows_service_account.email}"]
-
-  depends_on = [google_service_account.workflows_service_account]
+  role     = "roles/iam.serviceAccountTokenCreator"
+  member    = "serviceAccount:service-${data.google_project.project.number}@gcp-sa-pubsub.iam.gserviceaccount.com"
+  depends_on = [google_project_service.pubsub]
```

Comment:
Did you test this? I seem to remember you weren't able to put bindings dependent on something else but maybe I just did it wrong then..

Reply:
The dependency is indeed not tested